types/logger/logger_test.go

```diff
@@ -19,3 +20,19 @@ func TestStdLogger(t *testing.T) {
 	lg := StdLogger(t.Logf)
 	lg.Printf("plumbed through")
 }
+
+func TestRateLimiter(t *testing.T) {
```

Comment:
You probably want to use a fake Now() function for tests. To make that work, you need the rate limiter to accept a Now function at construction time (which in normal use is time.Now). Look at oss/control/controlcient/direct.go (and its use of a timeNow function) for an example.

Otherwise, this test could be unreliable if it runs slower than expected (like on a busy VM). Using a fake time function also lets you avoid using time.Sleep() in tests, which you should never do (if everyone did that, our tests would run much more slowly than they already do). You can instead have your fake timeNow function advance its timer, pretending to be 200ms in the future.

Reply:
With the LRU this ratelimit function doesn't use current time for anything anymore, everything time related is hidden in the `rate` implementation :)